### PR TITLE
Replace new UTF8Encoding(false)

### PR DESCRIPTION
**Pull request by nxtn**
... with `Encoding.Default` since they are the same.

https://github.com/dotnet/runtime/blob/436284b303a5b92eac38b4b66ae1b490b8fd1b48/src/libraries/System.Private.CoreLib/src/System/Text/Encoding.cs#L80

### CODE REVIEW

**Comment by eiriktsarpalis** [comment]
Thank you for your contribution.

If I may ask, what is the motivation for introducing this PR? While it is true that `Encoding.Default` happens to have the same configuration as the currently used encoding, how does the change clarify the code or improve performance? (apart from eliminating one allocation).

Note that `Encoding.Default` actually returns an instance of [Utf8EncodingSealed](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/Text/UTF8Encoding.Sealed.cs#L15). Seems like it's functionally equivalent to the base class (although I defer to @GrabYourPitchforks on that). So perhaps the devirtualization benefits alone might justify this change.

File: src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs [hunk 1]
```diff
@@ -16,8 +16,6 @@ namespace System.Diagnostics
 {
     public partial class Process : IDisposable
     {
-        private static readonly UTF8Encoding s_utf8NoBom =
-            new UTF8Encoding(encoderShouldEmitUTF8Identifier: false);
```
Comment (eiriktsarpalis) [inline src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs:20]: The problem with this change is that you're now specifying the default IO encoding in three separate locations. You might want to do something like
```csharp
private static readonly UTF8Encoding s_defaultEncoding = Encoding.Default;
```

**Comment by NextTurn** [comment]
It's just to eliminate the allocations, I think.

**Comment by huoyaoyuan** [comment]
> It's just to eliminate the allocations, I think.

The allocations are static, not instance. No need to remove.

To be meaningful, there may be new static instances for `UTF8BOM` and `UTF8NoBOM`.

File: src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs [hunk 2]
```diff
@@ -16,8 +16,7 @@ namespace System.Diagnostics
...
+        private static readonly UTF8Encoding s_utf8NoBom = Encoding.Default;
```
Comment (stephentoub) [inline src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs:19]: We don't need the field at all... the three call sites in this file can just use Encoding.Default directly.

**Comment by NextTurn** [comment]
> The allocations are static, not instance. No need to remove.

In this PR, yes. What do you think of those non-cached allocations among the libraries?

https://github.com/dotnet/runtime/blob/f81e70863c7ae5da7b4c8bbeb7297e8c8cf43572/src/libraries/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/CanonicalXml.cs#L110

**Comment by stephentoub** [comment]
> What do you think of those non-cached allocations among the libraries?

It is not worth complicating that CanonicalXml code to remove the UTF8Encoding allocation, as a) the code would need to be complicated to differentiate Windows from Unix, b) that Encoding allocation is dwarfed by the rest of the allocation happening in that method and its consumers, and c) is not on a hot path.

**Review by eiriktsarpalis — APPROVED** [review]

**Review by stephentoub — APPROVED** [review]

**Comment by GrabYourPitchforks** [comment]
Not an issue in this PR, but in general, `Encoding.Default` may be different across different runtimes. For example, on .NET Framework it's usually initialized to the system code page and isn't guaranteed to be UTF-8. Since all of the affected packages in this PR target only netcore, we're ok.

In general I'm a fan of readability / understandability, even at the expense of a single allocation. If the code _requires_ UTF-8 for correctness, it should be hardcoded against an API that has "utf8" somewhere in the API name.

**Pull request merged**